### ISSUE TITLE
Hello, there seems to be a slight error in sample 2.1

### ISSUE THREAD

**Issue by shuailiu199966** [issp-center-dev/HPhi#163]
Hello, I have three questions. 
Based on the code in sample 2.1, the entropy value that I obtained is negative and it is the same for all models.
Can the Python script you provided be used for calculations on all models？
How should I calculate the average bond density（W_p）?

**Comment by shuailiu199966** [comment]
![微信图片_20240514152321](https://github.com/issp-center-dev/HPhi/assets/148251279/ada6d789-8001-4adb-916e-4d0cb1425fa1)

**Comment by tmisawa** [comment]
This problem is caused by the fact that the entropy has degrees of freedom in constant shifts. In this calculation, the high-temperature limit of entropy is set to zero 
 and the zero-temperature limit of the entropy is `-log(d_H)`, where d_H is the dimensions of the Hilbert space. For example, if you take the 12-site spin 1/2 model, d_H is given by `2^{12}`. In that case, the entropy converges to `-log(2^12)~ -8.3177`.  If you want to set `S(T=0)=0`, please add `log(d_H)` to the calculation results.

**Comment by shuailiu199966** [comment]
&nbsp;If I set L=2 and W=3 for the 2D kaitev model under the standard model, then should d_H be 2 ^ {6} or 2 ^ {12}? What other parameters should I change
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Tue, May 14, 2024 03:36 PM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [issp-center-dev/HPhi] Hello, there seems to be a slight error in sample 2.1 (Issue #163)

&nbsp;



 
This problem is caused by the fact that the entropy has degrees of freedom in constant shifts. In this calculation, the high-temperature limit of entropy is set to zero
 and the zero-temperature limit of the entropy is -log(d_H), where d_H is the dimensions of the Hilbert space. For example, if you take the 12-site spin 1/2 model, d_H is given by 2^{12}. In that case, the entropy converges to -log(2^12)~ -8.3177.  If you want to set S(T=0)=0, please add log(d_H) to the calculation results.

**Comment by tmisawa** [comment]
In that case, `d_H=2^12`. You can find the information of `d_H` in standard output by HPhi.
For example,  `MAX DIMENSION idim_max=4096`. Here, `idim_max` means `d_H`.

**Comment by tmisawa** [comment]
You can also find information of `idim_max` in ` output/CHECK_Memory.dat`.

**Comment by tmisawa** [comment]
>How should I calculate the average bond density（W_p）?


Does W_p mean the flux of the Kitaev model? If so, it is possible to calculate W_p.
Since W_p is defined by the product of six spin operators, i.e., `W_p=Sx*Sy*Sz*Sx*Sy*Sz`,
it is necessary to an input file for calculating six-body Green functions.   
Please note that `SixBodyG` is the keyword for specifying the six-body Green functions in `calmod.def` .

**Comment by shuailiu199966** [comment]
> This problem is caused by the fact that the entropy has degrees of freedom in constant shifts. In this calculation, the high-temperature limit of entropy is set to zero and the zero-temperature limit of the entropy is `-log(d_H)`, where `d_H` is the dimensions of the Hilbert space. For example, if you take the 12-site spin 1/2 model, `d_H` is given by `2^{12}`. In that case, the entropy converges to `-log(2^12)~ -8.3177` at the zero-temperature limit. If you want to set `S(T=0)=0`, please add `log(d_H)` to the calculation results.



> &nbsp;如果我在标准模型下为 2D kaitev 模型设置 L=2 和 W=3，那么d_H应该是 2 ^ {6} 还是 2 ^ {12}？我还应该更改哪些其他参数 &nbsp; ------------------&nbsp;原版&nbsp;------------------ 发件人： ***@***.***>; 日期： &nbsp;周二， 2024年5月14日 03：36 PM 收件人：***@***.***>; 抄送：***@***.***>;@***.***>; 主题： &nbsp;回复：[issp-center-dev/HPhi] 您好，示例 2.1（问题 #163）中似乎存在轻微错误 &nbsp; 这个问题是由熵在恒定位移中具有自由度这一事实引起的。在此计算中，熵的高温极限设置为零 熵的零温度极限是 -log（d_H），其中 d_H 是希尔伯特空间的维数。例如，如果采用 12 位自旋 1/2 模型，则d_H由 2^{12} 给出。在这种情况下，熵收敛到 -log（2^12）~ -8.3177。如果要设置 S（T=0）=0，请在计算结果中添加 log（d_H）。 — 直接回复此电子邮件，在 GitHub 上查看或取消订阅。 您收到此消息是因为您创作了线程。消息 ID：***@***.***>

Hello,In your example, this formula { (log (Z)+<H>/T)/log (dH)}  was used to calculate the entropy of the Heisenberg model. Can this formula be used for other models, such as kitaev. It is very sweet of you!!!

**Comment by tmisawa** [comment]
Yes. This formula can be used for any other model, such as the Kitaev model. This factor comes from the normalization of wave functions and does not relate to lattice geometry.
